### PR TITLE
Add some smartness to what to build in PR's.

### PR DESCRIPTION
**Pull request by aressem**

### CODE REVIEW

File: travis/travis-build.sh [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    make -j ${NUM_THREADS}
+    ctest3 --output-on-failure -j ${NUM_THREADS}
+    ccache --show-stats
+    ;;
+  java)
+    env VESPA_MAVEN_EXTRA_OPTS="--no-snapshot-updates --batch-mode --threads ${NUM_THREADS}" sh ./bootstrap.sh java
+    mvn -V install --no-snapshot-updates --batch-mode --threads ${NUM_THREADS}
+    ;;
+  *)
+    env VESPA_MAVEN_EXTRA_OPTS="--no-snapshot-updates --batch-mode --threads ${NUM_THREADS}" sh ./bootstrap.sh java
```
Comment (hmusum) [inline travis/travis-build.sh:39]: Shouldn't this be `bootstrap.sh full`?

**Review by hmusum — CHANGES_REQUESTED** [review]
One comment that I think should be addressed, otherwise looks OK to me.

File: travis/travis-build.sh (hunk 1 above)
Comment (aressem) [reply travis/travis-build.sh:39]: When we build all, all the java artifacts will be built with Maven.  Using `full` here will work but the modules will be recompiled (waste) on the next line.

Comment (hmusum) [reply travis/travis-build.sh:39]: Great!

**Review by hmusum — APPROVED** [review]

**Pull request merged**